packages/ember-routing/tests/system/controller_for_test.js

```diff
@@ -1,112 +1,59 @@
+  ['@test controllerFor should lookup for registered controllers'](assert) {
+    this.add('controller:app', Controller.extend());
 
-  owner.__registry__.resolver = resolverFor(namespace);
-  owner.registerOptionsForType('view', { singleton: false });
-
-  owner.register('application:main', namespace, { instantiate: false });
-
-  owner.register('controller:basic', Controller, { instantiate: false });
-
-  return owner;
-}
-
-function resolverFor(namespace) {
-  return {
-    resolve(fullName) {
-      let nameParts = fullName.split(':');
-      let type = nameParts[0];
-      let name = nameParts[1];
-
-      if (name === 'basic') {
-        name = '';
-      }
-      let className = StringUtils.classify(name) + StringUtils.classify(type);
-      let factory = get(namespace, className);
-
-      if (factory) { return factory; }
-    }
-  };
-}
-
-let appInstance, appController, namespace;
-
-QUnit.module('Ember.controllerFor', {
-  setup() {
-    namespace = Namespace.create();
-    appInstance = buildInstance(namespace);
-    appInstance.register('controller:app', Controller.extend());
-    appController = appInstance.lookup('controller:app');
-  },
-  teardown() {
-    run(() => {
-      appInstance.destroy();
-      namespace.destroy();
+    return this.visit('/').then(() => {
+      let appInstance = this.applicationInstance;
+      let appController = appInstance.lookup('controller:app');
+      let controller = controllerFor(appInstance, 'app');
+      assert.equal(appController, controller, 'should find app controller');
     });
   }
 });
 
-QUnit.test('controllerFor should lookup for registered controllers', function() {
-  let controller = controllerFor(appInstance, 'app');
-
-  equal(appController, controller, 'should find app controller');
-});
-
-QUnit.module('Ember.generateController', {
-  setup() {
-    namespace = Namespace.create();
-    appInstance = buildInstance(namespace);
-  },
-  teardown() {
-    run(() => {
-      appInstance.destroy();
-      namespace.destroy();
+moduleFor('Ember.generateController', class extends ApplicationTestCase {
+  ['@test generateController should return Ember.Controller'](assert) {
+    return this.visit('/').then(() => {
+      let controller = generateController(this.applicationInstance, 'home');
+      assert.ok(controller instanceof Controller, 'should return controller');
     });
   }
-});
 
-QUnit.test('generateController should return Ember.Controller', function() {
-  let controller = generateController(appInstance, 'home');
+  ['@test generateController should return App.Controller if provided'](assert) {
+    let MainController = Controller.extend();
+    this.add('controller:basic', MainController);
 
-  ok(controller instanceof Controller, 'should return controller');
-});
-
-
-QUnit.test('generateController should return App.Controller if provided', function() {
-  let controller;
-  namespace.Controller = Controller.extend();
```

Comment:
Seems like it is `basic` https://github.com/emberjs/ember.js/blob/ee8ab348484421f254fe995c6f04f0d4a0859cfa/packages/ember-routing/lib/system/generate_controller.js#L19 Works in the resolver b/c `useRouterNaming` is called for controllers https://github.com/emberjs/ember.js/blob/master/packages/ember-application/lib/system/resolver.js#L288 barf.